great_expectations/datasource/data_connector/util.py

```diff
@@ -290,6 +288,49 @@ def get_filesystem_one_level_directory_glob_path_list(
     return path_list
 
 
+def list_azure_keys(
+    azure: BlobServiceClient,
+    query_options: dict,
+    recursive: bool = False,
+) -> List[str]:
+    """
+    Utilizes the Azure Blob Storage connection object to retrieve blob names based on user-provided criteria.
+
+    For InferredAssetAzureDataConnector, we take container and name_starts_with and search for files using RegEx at and below the level
```

Comment:
Are `container` and `name_starts_with` already in `base.py` (in the appropriate schemas), or not yet (should they be)?

Reply:
yes! They've been added in a prior PR: https://github.com/great-expectations/great_expectations/pull/3188/files